assets/js/modules/analytics-4/utils/data-mock.js

```diff
@@ -0,0 +1,468 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import md5 from 'md5';
+import faker from 'faker';
+import invariant from 'invariant';
+import castArray from 'lodash/castArray';
+import { zip, from, Observable } from 'rxjs';
+import { map, reduce, take } from 'rxjs/operators';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * Internal dependencies
+ */
+import { MODULES_ANALYTICS_4 } from '../datastore/constants';
+import { isValidDateString } from '../../../util';
+import { stringToDate } from '../../../util/date-range/string-to-date';
+
+const ANALYTICS_4_METRIC_TYPES = {
+	totalUsers: 'INTEGER',
+	newUsers: 'INTEGER',
+	sessions: 'INTEGER',
+	conversions: 'INTEGER',
+	screenPageViews: 'INTEGER',
+	engagedSessions: 'INTEGER',
+	averageSessionDuration: 'TIME',
+};
+
+const ANALYTICS_4_DIMENSION_OPTIONS = {
+	sessionDefaultChannelGrouping: [
+		'Direct',
+		'Organic Search',
+		'Paid Social',
+		'Organic Social',
+		'Email',
+		'Affiliates',
+		'Referral',
+		'Paid Search',
+		'Video',
+		'Display',
+	],
+	country: [
+		'United States',
+		'United Kingdom',
+		'India',
+		'(not set)',
+		'France',
+		'Ukraine',
+		'Italy',
+		'Mexico',
+	],
+	deviceCategory: [ 'Desktop', 'Tablet', 'Mobile' ],
+	pageTitle: ( i ) => ( i <= 12 ? `Test Post ${ i }` : false ),
+	pagePathPlusQueryString: ( i ) =>
+		i <= 12 ? `/test-post-${ i }/` : false,
+};
+
+/**
+ * Gets metric key.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|Object} metric Metric name or object.
+ * @return {string} Metric key.
+ */
+function getMetricKey( metric ) {
+	return metric?.expression || metric.toString();
+}
+
+/**
+ * Gets metric type.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|Object} metric Metric name or object.
+ * @return {string} Type of the metric.
+ */
+function getMetricType( metric ) {
+	return ANALYTICS_4_METRIC_TYPES[ getMetricKey( metric ) ];
+}
+
+/**
+ * Generates and returns metric values.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<Object>} validMetrics Metric list.
+ * @return {Array.<Object>} Array of metric values.
+ */
+function generateMetricValues( validMetrics ) {
+	validMetrics.forEach( ( validMetric ) => {
+		switch ( getMetricType( validMetric ) ) {
+			case 'INTEGER':
+				return [
+					{
+						value: faker.datatype
+							.number( { min: 0, max: 100 } )
+							.toString(),
+					},
+				];
+
+			case 'PERCENT':
+				return [
+					{
+						value: faker.datatype
+							.float( { min: 0, max: 100 } )
+							.toString(),
+					},
+				];
+			case 'TIME':
+				return [
+					{
+						value: faker.datatype
+							.number( { min: 0, max: 3600 } )
+							.toString(),
+					},
+				]; // 1 hour max.
+			case 'CURRENCY':
+				return [
+					{
+						value: faker.datatype
+							.float( { min: 0, max: 10000 } )
+							.toString(),
+					},
+				]; // $10k max.
+		}
+	} );
+
+	return [];
```

Comment:
Ah, this function is now hardwired to return an empty array - this is why `metricValues` is empty as per my other comment.

Reply:
👍